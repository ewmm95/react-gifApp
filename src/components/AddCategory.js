import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = ({target: {value}}) => setInputValue(value)
    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length <= 2) return;
        setCategories( x => [inputValue, ...x] );
        setInputValue('');
    }
    return (
        <form onSubmit={handleSubmit}>
            <input
                className='input-search'
                type='text'
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}