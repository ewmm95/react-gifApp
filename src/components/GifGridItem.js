import React from 'react';
import 'animate.css';
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {
    return (
        <div className='card animate__animated animate__zoomIn' tabIndex="0">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}