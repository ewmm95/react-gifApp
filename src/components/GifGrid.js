import React, { useState, useEffect } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import 'animate.css';

export const GifGrid  = ({category}) => {
    const { data: images, loading} = useFetchGifs(category);
    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>
            {loading && <h3  className='animate__animated animate__flash'>Cargando...</h3>}
            <div className='cards'>
                { images.map( img => <GifGridItem key={img.id} {...img}/>) }
            </div>
        </>
    )
}
