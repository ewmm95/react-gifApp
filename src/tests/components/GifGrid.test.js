import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Pruebas en <GifGrid />', () => {
    const props = {
        category: 'categoria'
    };
    test('debe mostrar el componente correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid {...props}/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe mostrar los items cuando se cargan las imagenes', () => {
        const gifs = [{
            id: 'id',
            title: 'Titulo',
            url: 'https://localhost:180908/login',
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper = shallow(<GifGrid {...props}/>);
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
    
})
