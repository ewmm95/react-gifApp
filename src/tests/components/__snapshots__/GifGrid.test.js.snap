// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    categoria
  </h3>
  <h3
    className="animate__animated animate__flash"
  >
    Cargando...
  </h3>
  <div
    className="cards"
  />
</Fragment>
`;
